### ISSUE TITLE
After upgrading to 1.17.0 kubectl doesn't work on master

### ISSUE THREAD

**Issue by shachar-ash** [kubernetes/kops#9358]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

> Version 1.17.0 (git-a17511e6dd)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

> Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-21T14:51:23Z", GoVersion:"go1.14.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.6", GitCommit:"d32e40e20d167e103faf894261614c5b45c44198", GitTreeState:"clean", BuildDate:"2020-05-20T13:08:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

**3. What cloud provider are you using?**
AWS
**4. What commands did you run?  What is the simplest way to reproduce this issue?**
I just created a new cluster using kops 1.17.0. Running kubectl from clients works, but when connecting to the master instance all kubectl commands failed.
**5. What happened after the commands executed?**

> ubuntu@kube-master:~ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Error from server (NotFound): the server could not find the requested resource
ubuntu@kube-master:~ kubectl get nodes
error: the server doesn't have a resource type "nodes"

**6. What did you expect to happen?**

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

> apiVersion: kops/v1alpha2
> kind: Cluster
> metadata:
>   creationTimestamp: null
>   name: kube.dp.int.sergio.net
> spec:
>   api:
>     dns: {}
>   authorization:
>     rbac: {}
>   channel: stable
>   cloudLabels:
>     env_id: prod
>     system: dp
>     accounting: prod
>   cloudProvider: aws
>   configBase: s3://my-kops-backend/kube.dp.int.sergio.net
>   dnsZone: Z8SZ1W1WS0SUO
>   etcdClusters:
>   - etcdMembers:
>     - instanceGroup: master-us-west-1c
>       name: c
>     name: main
>   - etcdMembers:
>     - instanceGroup: master-us-west-1c
>       name: c
>     name: events
>   kubernetesApiAccess:
>   - 172.31.0.0/16
>   kubernetesVersion: v1.17.6
>   masterInternalName: api.internal.kube.dp.int.sergio.net
>   masterPublicName: api.kube.dp.int.sergio.net
>   networkCIDR: 172.31.0.0/16
>   networkID: vpc-12345
>   networking:
>     calico: {}
>   nonMasqueradeCIDR: 100.64.0.0/10
>   sshAccess:
>   - 172.31.0.0/16
>   subnets:
>   - cidr: 172.31.36.0/24
>     name: us-west-1c
>     type: Private
>     zone: us-west-1c
>   - cidr: 172.31.37.0/24
>     name: utility-us-west-1c
>     type: Utility
>     zone: us-west-1c
> 
>   topology:
>     dns:
>       type: Public
>     masters: private
>     nodes: private
> ---
> 
> apiVersion: kops/v1alpha2
> kind: InstanceGroup
> metadata:
>   creationTimestamp: null
>   labels:
>     kops.k8s.io/cluster: kube.dp.int.sergio.net
>   name: master-us-west-1c
> spec:
>   additionalSecurityGroups:
>   - sg-123456 
>   - sg-1234567 
>   image: ami-123456
>   machineType: m5.large
>   maxSize: 1
>   minSize: 1
>   nodeLabels:
>     kops.k8s.io/instancegroup: master-us-west-1c
>   cloudLabels:
>     Project_name: Analytics
>     Name: kube-master.dp-kubernetes-cluster.int.sergio.net
>     Role: dp-master
>   role: Master
>   rootVolumeSize: 75
>   subnets:
>   - us-west-1c
> 
> ---
> 
> apiVersion: kops/v1alpha2
> kind: InstanceGroup
> metadata:
>   creationTimestamp: null
>   labels:
>     kops.k8s.io/cluster: kube.dp.int.sergio.net
>   name: general
> spec:
>   iam:
>     profile: arn:aws:iam::12345:instance-profile/Kube-Cluster
>   additionalSecurityGroups:
>   - sg-123456 
>   - sg-1234567 
>   image: ami-123456
>   machineType: m5.4xlarge
>   maxSize: 3
>   minSize: 1
>   nodeLabels:
>     kops.k8s.io/instancegroup: general
>   cloudLabels:
>     Name: general-worker.dp-kubernetes-cluster.int.sergio.net
>     Role: general-worker
>   role: Node
>   rootVolumeSize: 150
>   subnets:
>   - us-west-1c
> 

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

> I0614 08:20:28.050193    7150 round_trippers.go:423] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.18.3 (linux/amd64) kubernetes/2e7996e" 'http://localhost:8080/version?timeout=32s'
I0614 08:20:28.057660    7150 round_trippers.go:443] GET http://localhost:8080/version?timeout=32s 404 Not Found in 7 milliseconds
I0614 08:20:28.057675    7150 round_trippers.go:449] Response Headers:
I0614 08:20:28.057680    7150 round_trippers.go:452]     Date: Sun, 14 Jun 2020 08:20:28 GMT
I0614 08:20:28.057752    7150 round_trippers.go:452]     Content-Length: 10
I0614 08:20:28.057761    7150 round_trippers.go:452]     Content-Type: text/plain; charset=utf-8
I0614 08:20:28.057768    7150 round_trippers.go:452]     X-Content-Type-Options: nosniff
I0614 08:20:28.063077    7150 request.go:1068] Response Body: not found
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
I0614 08:20:28.065559    7150 helpers.go:216] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "the server could not find the requested resource",
  "reason": "NotFound",
  "details": {
    "causes": [
      {
        "reason": "UnexpectedServerResponse",
        "message": "not found"
      }
    ]
  },
  "code": 404
}]
F0614 08:20:28.066400    7150 helpers.go:115] Error from server (NotFound): the server could not find the requested resource

**9. Anything else do we need to know?**
This bug happens for me for all 1.17.X Kubernetes versions using kops 1.17.0.
I tried to create 1.6.10 cluster using kops 1.17.0 and it worked just fine.
At the beginning I used client version 1.17.6 on the master, I upgraded it to 1.18.3, all commands failed with both versions.

kubectl works as expected remotely - I'm able to access the server and do anything regularly, it only fails on the master instance.

I don't know where to look, any ideas?

**Comment by olemarkus** [comment]
I assume there is a "1" missing from those last version numbers there.
Which AMI are you using? Or which distro is it based on?

**Comment by hakman** [comment]
I guess this should work. Don't really know why it doesn't work by default.
```
 sudo kubectl --kubeconfig=/var/lib/kubectl/kubeconfig get nodes
```

**Comment by shachar-ash** [comment]
> I assume there is a "1" missing from those last version numbers there.
> Which AMI are you using? Or which distro is it based on?

There is definetely a "1" missing, edited it.
It's a custom AMI based on Ubuntu.



> I guess this should work. Don't really know why it doesn't work by default.
> 
> ```
>  sudo kubectl --kubeconfig=/var/lib/kubectl/kubeconfig get nodes
> ```

It did work, it doesn't solve the issue and I'm still looking for a more "native" solution but I might use it to create a workaround, thanks!

**Comment by hakman** [comment]
#9369 should take care of this issue.

**Comment by hakman** [comment]
Fixed by #9369. Will be available in Kops 1.17.1.
/close
